File: src/main/webapp/src/app/models/action-trigger-midi-system-real-time.ts
import { ActionTriggerMidi } from "./action-trigger-midi";

export class ActionTriggerMidiSystemRealTime extends ActionTriggerMidi {
  systemRealTimeType: string = "START";

  constructor(data?: any) {
    super(data);

    if (!data) {
      return;
    }

    this.systemRealTimeType = data.systemRealTimeType;

    if (!this.systemRealTimeType) {
      this.systemRealTimeType = "START";
    }
  }

  toJSON() {
    return { actionTriggerMidiSystemRealTime: { ...this } };
  }
}

File: src/main/java/com/ascargon/rocketshow/midi/ActionTriggerMidiSystemRealTime.java
package com.ascargon.rocketshow.midi;

import lombok.Getter;
import lombok.Setter;

/**
 * Triggered by one of the MIDI system real-time transport messages, which a master sends to run the
 * transport of everything on the MIDI line at once. They carry no channel, so the channel of the
 * trigger is ignored.
 */
@Getter
@Setter
public class ActionTriggerMidiSystemRealTime extends ActionTriggerMidi {

    public enum SystemRealTimeType {
        // Start playing from the beginning (0xFA)
        START,
        // Resume playing from the current position (0xFB)
        CONTINUE,
        // Stop playing (0xFC)
        STOP
    }

    private SystemRealTimeType systemRealTimeType;

}
